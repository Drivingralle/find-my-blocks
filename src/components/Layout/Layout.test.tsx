import React from "react";
import renderer from "react-test-renderer";
import { Layout } from "./";

jest.mock("../../hooks", () => ({
  useThrottledResizeObserver: jest
    .fn()
    .mockReturnValue([null, { width: 1, height: 1 }]),
}));

it("render correctly", () => {
  const tree = renderer.create(<Layout />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders all areas correctly", () => {
  const sidebar = renderer.create(<Layout sidebar="sidebar" />).toJSON();
  const settings = renderer.create(<Layout settings="settings" />).toJSON();
  const title = renderer.create(<Layout title="title" />).toJSON();
  const cards = renderer.create(<Layout cards="cards" />).toJSON();
  expect(sidebar).toMatchSnapshot();
  expect(settings).toMatchSnapshot();
  expect(title).toMatchSnapshot();
  expect(cards).toMatchSnapshot();
});
