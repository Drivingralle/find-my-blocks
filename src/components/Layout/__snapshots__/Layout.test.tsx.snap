// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="layout"
>
  <div
    className="main"
  >
    <div
      className="footer"
    >
      FOOTER
    </div>
  </div>
</div>
`;

exports[`renders all areas correctly 1`] = `
<div
  className="layout"
>
  <div
    className="sidebar"
  >
    sidebar
  </div>
  <div
    className="main"
  >
    <div
      className="footer"
    >
      FOOTER
    </div>
  </div>
</div>
`;

exports[`renders all areas correctly 2`] = `
<div
  className="layout"
>
  <div
    className="main"
  >
    <div
      className="box"
    >
      settings
    </div>
    <div
      className="footer"
    >
      FOOTER
    </div>
  </div>
</div>
`;

exports[`renders all areas correctly 3`] = `
<div
  className="layout"
>
  <div
    className="main"
  >
    <div
      className="box"
    >
      <h1
        className="heading h1"
      >
        title
      </h1>
    </div>
    <div
      className="footer"
    >
      FOOTER
    </div>
  </div>
</div>
`;

exports[`renders all areas correctly 4`] = `
<div
  className="layout"
>
  <div
    className="main"
  >
    <div
      className="box"
    >
      cards
    </div>
    <div
      className="footer"
    >
      FOOTER
    </div>
  </div>
</div>
`;
